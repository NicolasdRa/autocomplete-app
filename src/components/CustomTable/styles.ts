import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		container: {
			marginTop: '1rem',
		},

		pagination: {
			marginRight: '2rem',
		},

		title: {
			margin: '2rem 0 1rem 2rem',
		},
	}),
);
