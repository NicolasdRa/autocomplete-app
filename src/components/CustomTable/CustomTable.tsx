import React, { useState } from 'react';

import {
	Table,
	TableBody,
	TableHead,
	TableCell,
	TableContainer,
	TableRow,
	TablePagination,
} from '@material-ui/core';

import { CustomRow } from './CustomRow/CustomRow';
import { useStyles } from './styles';

interface Column {
	title: string;
	align: 'left' | 'right' | 'inherit' | 'center' | 'justify' | undefined;
}

const columns: Column[] = [
	{ title: '', align: 'left' },
	{ title: 'Title', align: 'left' },
	{ title: 'Id', align: 'left' },
	{ title: 'Gender', align: 'left' },
	{ title: 'Price', align: 'left' },
	{ title: 'Sale Price', align: 'left' },
];

interface Props {
	data: any[];
}

export const CustomTable: React.FC<Props> = ({ data }) => {
	const classes = useStyles();

	// for pagination
	const [page, setPage] = useState<number>(0);
	const [rowsPerPage, setRowsPerPage] = useState<number>(10);

	// handles pagination
	const handleChangePage = (e: unknown, newPage: number) => {
		setPage(newPage);
	};

	const handleChangeRowsPerPage = (
		e: React.ChangeEvent<HTMLInputElement>,
	) => {
		setRowsPerPage(parseInt(e.target.value, 10));
		setPage(0);
	};

	return (
		<TableContainer className={classes.container}>
			<Table aria-label='collapsible table'>
				<TableHead>
					<TableRow>
						{columns.map((column) => (
							<TableCell
								key={column.title}
								align={column.align}
								className={classes.title}>
								{column.title}
							</TableCell>
						))}
						<TableCell />
					</TableRow>
				</TableHead>
				<TableBody>
					{data
						.slice(
							page * rowsPerPage,
							page * rowsPerPage + rowsPerPage,
						)
						.map((rowData) => (
							<CustomRow key={rowData.gtin} data={rowData} />
						))}
				</TableBody>
			</Table>
			<TablePagination
				rowsPerPageOptions={[5, 10, 25, 100, 200]}
				component='div'
				count={data.length}
				rowsPerPage={rowsPerPage}
				page={page}
				onPageChange={handleChangePage}
				onRowsPerPageChange={handleChangeRowsPerPage}
			/>
		</TableContainer>
	);
};
