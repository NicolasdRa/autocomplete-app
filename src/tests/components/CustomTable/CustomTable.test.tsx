import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { CustomTable } from '../../../components/CustomTable/CustomTable';

const mockData = [
	{
		title: 'test object',
		gtin: '1',
		gender: 'unisex',
		price: '1 euro',
		sale_price: '0.5 euro',
		image_link: 'testImg.jpg',
		additional_image_link: 'testImg1.jpg, testImg2.jpg',
	},
];

describe('tests on <CustomTable/>', () => {
	const wrapper = mount(<CustomTable data={mockData} />);

	test('should render <CustomTable/> correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should render as many rows as items in data array', () => {
		expect(wrapper.find('CustomRow').length).toBe(mockData.length);
	});
});
