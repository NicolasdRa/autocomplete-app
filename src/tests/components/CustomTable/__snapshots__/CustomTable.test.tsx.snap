// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <CustomTable/> should render <CustomTable/> correctly 1`] = `
<div
  className="MuiTableContainer-root makeStyles-container-1"
>
  <table
    aria-label="collapsible table"
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head makeStyles-title-3 MuiTableCell-alignLeft"
          scope="col"
        />
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head makeStyles-title-3 MuiTableCell-alignLeft"
          scope="col"
        >
          Title
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head makeStyles-title-3 MuiTableCell-alignLeft"
          scope="col"
        >
          Id
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head makeStyles-title-3 MuiTableCell-alignLeft"
          scope="col"
        >
          Gender
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head makeStyles-title-3 MuiTableCell-alignLeft"
          scope="col"
        >
          Price
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head makeStyles-title-3 MuiTableCell-alignLeft"
          scope="col"
        >
          Sale Price
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      Array [
        <tr
          className="MuiTableRow-root makeStyles-root-4 MuiTableRow-hover"
          onClick={[Function]}
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body makeStyles-firstCell-5"
          >
            <img
              alt="test object"
              className="makeStyles-productImg-6"
              loading="lazy"
              src="testImg.jpg"
            />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body makeStyles-secondCell-7"
          >
            <p
              className="MuiTypography-root makeStyles-primaryInfo-8 MuiTypography-body1"
            >
              test object
            </p>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <p
              className="MuiTypography-root makeStyles-primaryInfo-8 MuiTypography-body1"
            >
              1
            </p>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <p
              className="MuiTypography-root makeStyles-primaryInfo-8 MuiTypography-body1"
            >
              unisex
            </p>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <p
              className="MuiTypography-root makeStyles-primaryInfo-8 MuiTypography-body1"
            >
              1 euro
            </p>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <p
              className="MuiTypography-root makeStyles-primaryInfo-8 MuiTypography-body1"
            >
              0.5 euro
            </p>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <button
              aria-label="expand row"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>,
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            colSpan={12}
            style={
              Object {
                "padding": 0,
              }
            }
          />
        </tr>,
      ]
    </tbody>
  </table>
  <div
    className="MuiTablePagination-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
    >
      <div
        className="MuiTablePagination-spacer"
      />
      <p
        className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        id="mui-58065"
      >
        Rows per page:
      </p>
      <div
        className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        onClick={[Function]}
      >
        Array [
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-58065 mui-77683"
            className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            id="mui-77683"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            10
          </div>,
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value={10}
          />,
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>,
          "",
        ]
      </div>
      <p
        className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
      >
        1-1 of 1
      </p>
      <div
        className="MuiTablePagination-actions"
      >
        <button
          aria-label="Previous page"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Previous page"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Next page"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Next page"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
